neuralpp/symbolic/interval.py

```diff
@@ -0,0 +1,75 @@
+from __future__ import annotations
+from abc import abstractmethod
+from typing import Iterable, Callable
+from constants import if_then_else
+
+from .expression import Variable, Expression, Context
+from .z3_expression import Z3SolverExpression
+
+
+class ClosedInterval:
+    """ [lower_bound, upper_bound] """
```

Comment:
Since we're using `@property`, we might need an init function to help store self._lower_bound or self._upper_bound.

Reply:
Yes. We'll implement that here. Now it's just for interface discussion.